arch/xtensa/include/arch.h

```diff
@@ -91,6 +91,18 @@ extern "C"
 #define EXTERN extern
 #endif
 
+#ifdef CONFIG_XTENSA_USE_SEPERATE_IMEM
+
+struct mallinfo; /* Forward reference, see malloc.h */
+
+void      up_imm_initialize(void);
+FAR void *up_imm_malloc(size_t size);
+void      up_imm_free(FAR void *mem);
+FAR void *up_imm_memalign(size_t alignment, size_t size);
+bool up_imm_heapmember(FAR void *mem);
+int up_imm_mallinfo(FAR struct mallinfo *info);
+#endif
+
```

Comment:
The naming up_imm_* does not follow the NuttX naming conventions.  These should all be xtensa_imm_* .  See https://cwiki.apache.org/confluence/display/NUTTX/Naming+of+Architecture%2C+MCU%2C+and+Board+Interfaces